webapp/src/dogma/features/repo/RepoTokenList.tsx

```diff
@@ -0,0 +1,48 @@
+import { DeleteIcon } from '@chakra-ui/icons';
+import { Button, Badge } from '@chakra-ui/react';
+import { ColumnDef, createColumnHelper } from '@tanstack/react-table';
+import { DateWithTooltip } from 'dogma/common/components/DateWithTooltip';
+import { DynamicDataTable } from 'dogma/common/components/table/DynamicDataTable';
+import { RepoTokenDetailDto } from 'dogma/features/repo/RepoTokenDto';
+
+export type RepoTokenListProps<Data extends object> = {
+  data: Data[];
+};
+
+const RepoTokenList = <Data extends object>({ data }: RepoTokenListProps<Data>) => {
+  const columnHelper = createColumnHelper<RepoTokenDetailDto>();
+  const columns = [
+    columnHelper.accessor((row: RepoTokenDetailDto) => row.appId, {
+      cell: (info) => info.getValue(),
+      header: 'App ID',
+    }),
+    columnHelper.accessor((row: RepoTokenDetailDto) => row.role, {
+      cell: (info) => (
+        <Badge colorScheme={info.getValue().toLowerCase() === 'owner' ? 'blue' : 'green'}>
+          {info.getValue()}
+        </Badge>
+      ),
+      header: 'Role',
+    }),
+    columnHelper.accessor((row: RepoTokenDetailDto) => row.creation.user, {
+      cell: (info) => info.getValue(),
+      header: 'Added By',
+    }),
+    columnHelper.accessor((row: RepoTokenDetailDto) => row.creation.timestamp, {
+      cell: (info) => <DateWithTooltip date={info.getValue()} />,
+      header: 'Added At',
+    }),
+    columnHelper.accessor((row: RepoTokenDetailDto) => row.appId, {
+      cell: () => (
+        <Button leftIcon={<DeleteIcon />} size="sm" colorScheme="red">
+          Delete
+        </Button>
```

Comment:
Question: The legacy UI is able to change a role on the tokens page. Are you planning to add an edit button in future work?

Reply:
Let's add them when we're doing update/delete for this section.